packages/core/src/browser/common-frontend-contribution.ts

```diff
@@ -1446,10 +1451,43 @@ export class CommonFrontendContribution implements FrontendApplicationContributi
 
                 })
         ];
+
+        const CREATE_NEW_FILE_ITEM_ID = 'create-new-file';
+        const hasNewFileHandler = this.commandRegistry.getActiveHandler(CommonCommands.NEW_FILE.id) !== undefined;
+        // Create a "Create New File" item only if there is a NEW_FILE command handler.
+        const createNewFileItem: QuickPickItem & { value?: string } | undefined = hasNewFileHandler ? {
+            id: CREATE_NEW_FILE_ITEM_ID,
+            label: nls.localizeByDefault('Create New File ({0})'),
+            description: nls.localizeByDefault('Built-in'),
+            execute: async () => {
+                if (createNewFileItem?.value) {
+                    const parent = await this.workingDirProvider.getUserWorkingDir();
+                    // Exec NEW_FILE command with the file name as additional argument
+                    return this.commandRegistry.executeCommand(CommonCommands.NEW_FILE.id, parent, createNewFileItem.value);
```

Comment:
nit: I think that when it comes to parameters for this command the fileName is the more important and the mandatory one whereas the parent directory is more of an improved user experience so the fileName should come first and the potential parent should be second, what do you think?

Reply:
Just to explain my motivation:  There is a `getSelection()` function in `filesystem-frontend-contribution.ts` which is used in some other handlers. The first argument is a current selected file URI. First I was about to re-use this function and then implemented an own and just kept the order. Another motivation is, that a file name is not enough, one also needs to know where to create this file, so the parent is kind of mandatory too.
What I'm really missing is a command parameter definition to expand the `...args` parameter in something meaningful. Like e.g. in vs-code API or many other command APIs or UI frameworks.

However, I will change the order if it feels more natural.